aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md

```diff
@@ -1241,15 +1241,22 @@ In JavaScript, use an array buffer or a readable stream to receive the data:
 In C# code:
 
 ```csharp
-using var streamRef = new DotNetStreamReference(stream: {STREAM}, leaveOpen: false);
+var streamRef = new DotNetStreamReference(stream: {STREAM}, leaveOpen: false);
 await JS.InvokeVoidAsync("streamToJavaScript", streamRef);
 ```
 
 In the preceding example:
 
-* The `{STREAM}` placeholder represents the <xref:System.IO.Stream> sent to JavaScript.
+* The `{STREAM}` placeholder represents the <xref:System.IO.Stream> sent to JS.
 * `JS` is an injected <xref:Microsoft.JSInterop.IJSRuntime> instance.
 
+The `streamRef` varaible isn't declared with the `using` keyword because it would result in disposing the stream twice, once when transmission completes and again when `streamRef` exits the scope that it was declared in. Use the `using` keyword to manually dispose of the stream if ***either*** of the following conditions are met:
+
+* `leaveOpen` is set to `true` and .NET code waits for the stream to get completely consumed by JS before exiting the current scope. For example, it's valid to set `leaveOpen` to `true` and await a JS method invocation that returns a promise that completes only after the stream is completely consumed.
+* The `DotNetStreamReference` isn't passed to JS via JS interop.
+
+In most cases, `leaveOpen` is `false`, which is the default value, and the `DotNetStreamReference` is passed to JS, so you generally shouldn't manually dispose the `DotNetStreamReference`.
+
```

Comment:
Looking this over again, I think we can simplify the guidance a bit further:
1. I realized that disposing a `DotNetStreamReference` is actually a no-op if `leaveOpen` is true (so we can remove the bit about disposing when `leaveOpen` is true).
2. We should suggest not creating a `DotNetStreamReference` at all unless it's passed to a JS interop invocation. I can't think of a valid pattern that requires this, as you can always create the `DotNetStreamReference` just before passing it to a JS interop call.

But then I was also thinking that it could be beneficial to expand on _why_ developers should be cautious about disposing `DotNetStreamReference`s.

So maybe we could instead say something like this:

> It's almost never necessary to dispose a `DotNetStreamReference` instance. When `leaveOpen` is set to its default value of `false`, the underlying stream gets automatically disposed after being transmitted to JS.

> If `leaveOpen` is `true`, then disposing a `DotNetStreamReference` does not dispose its `stream`. It's up to the application to decide when to dispose the underlying stream. In doing so, it's important to consider the following:
> 1. Disposing a stream while it's being transmitted to JS is considered an application error and may cause the circuit to get terminated or the WebAssembly app to crash.
> 2. Stream transmission begins as soon as the `DotNetStreamReference` gets passed as an argument to a JS interop call, regardless of whether the stream actually gets used in JS logic.

> Given these characteristics, we recommend disposing the underlying stream _only after it gets fully consumed by JS_ (i.e., the promise returned by `.arrayBuffer()` or `.stream()` resolves). It follows that a `DotNetStreamReference` should only get passed to JS if it is unconditionally going to be consumed by JS logic.

(feel free to adjust the wording to sounds more docs-ey 🙂)